include/xtensor/xcontainer.hpp

```diff
@@ -220,7 +190,7 @@ namespace xt
         using inner_strides_type = typename base_type::inner_strides_type;
         using inner_backstrides_type = typename base_type::inner_backstrides_type;
 
-        void reshape(const shape_type& shape);
+        void reshape(const shape_type& shape, bool force = false);
         void reshape(const shape_type& shape, layout l);
         void reshape(const shape_type& shape, const strides_type& strides);
```

Comment:
I think one benefit of `static_asserts` would be that we could print a helpful error message, e.g. "reshaping with layout parameter only possible if layout == dynamic". 
I have changed it now, though, to disable reshaping with layout if L != dynamic.